File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        depth = 0
        ans = 0
        q = [(root, 0)]
        
        while len(q) != 0:
            v, d = q.pop(0)

            if d > depth:
                depth = d
                ans = 0
            ans += v.val

            if v.left is not None:
                q.append((v.left, d + 1))
            if v.right is not None:
                q.append((v.right, d + 1))
                
        return ans

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        int depth = 0;
        int ans = 0;
        queue<pair<TreeNode*, int>> q;
        
        if (root == nullptr)
            return 0;

        q.push({root, 0});

        while (!q.empty()) {
            auto [node, d] = q.front();
            q.pop();

            if (d > depth) {
                depth = d;
                ans = 0;
            }
            ans += node->val;

            if (node->left) q.push({node->left, d + 1});
            if (node->right) q.push({node->right, d + 1});
        }
        return ans;
    }
};